infra/api/service/main.tf

```diff
@@ -95,8 +95,9 @@ data "aws_rds_cluster" "db_cluster" {
 }
 
 data "aws_acm_certificate" "cert" {
-  count  = local.service_config.domain_name != null ? 1 : 0
-  domain = local.service_config.domain_name
+  count       = local.service_config.domain_name != null ? 1 : 0
+  domain      = local.service_config.domain_name
+  most_recent = true
```

Comment:
I just found `most_recent` while looking into something else

It's actually tangential to the main point of the PR, but... I figured since I'm messing with certs anyways

docs: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/acm_certificate#most_recent-1